tests/fixtures/bin/eslint.config-promise-tick-throws.js

```diff
@@ -0,0 +1,13 @@
+function throwError() {
+    const error = new Error();
+    error.stack = "test_error_stack";
+    throw error;
+}
+
+process.nextTick(throwError);
+
+/*
+ * Promise ensures that this config must be await-ed and therefore
+ * the error in next tick will be thrown before any linting is done
+ */
+module.exports = Promise.resolve([{}]);
```

Comment:
Just a note that using `Promise.resolve` shouldn't be necessary. In the current implementation, [`loadFlatConfigFile`](https://github.com/eslint/eslint/blob/0a9c43339a4adef24ef83034d0b078dd279cc977/lib/eslint/flat-eslint.js#L256-L304) always returns a Promise, because it's an async function. It would be pretty tricky to make it return synchronously, and that would be only possible for CommonJS modules.